src/actions/users.ts

```diff
@@ -779,6 +779,69 @@ export function setStatus(status: UserStatus): ActionFunc {
     };
 }
 
+export function setCustomStatus(customStatus: UserCustomStatus): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let data;
+        try {
+            await Client4.updateCustomStatus(customStatus);
+            data = await Client4.getMe();
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        dispatch({
+            type: UserTypes.RECEIVED_ME,
+            data,
+        });
+        dispatch(loadRolesIfNeeded(data.roles.split(' ')));
```

Comment:
Do we need to call load roles here?

I wonder if you can just use `bindClientFunc` instead.

Reply:
load roles is called because it is being called everywhere if `UserTypes.RECEIVED_ME` action is dispatched.
And we can use bindClientFunc but I don't think using it is necessary here.